autochain/models/chat_openai.py

```diff
@@ -230,3 +230,129 @@ def _create_llm_result(self, response: Mapping[str, Any]) -> LLMResult:
         llm_output = {"token_usage": response["usage"], "model_name": self.model_name}
         result = LLMResult(generations=generations, llm_output=llm_output)
         return result
+
+
+class ChatAzureOpenAI(BaseLanguageModel):
```

Comment:
let's combine this into the same ChatOpenAI model. 
we would need to add the following to the model 
if api type is azure, validate all other information existed set them to openai at `validate_environment` step. 
```
api_type: Optional[str] = None
azure_api_base: Optional[str] = None
azure_api_verision : Optional[str] = None
azure_deployment_name: : Optional[str] = None
```

Reply:
Thanks for your suggestions.

I have a point to discuss.

We know that both `api_version` and 
`deployment_name` exist for Azure, so adding `azure_` prefix is ​​normal. 

But for `api_base`, if the user uses a proxy to access OpenAI, the value can also be modified.

I think we can use `api_base` like `api_type` or use `openai_api_base`

WDYT？

Reply:
makes sense. please put some comments for each parameter